xbmc/windowing/gbm/WinSystemGbmGLESContext.cpp

```diff
@@ -33,6 +37,10 @@
 
 using namespace KODI;
 
+CWinSystemGbmGLESContext::CWinSystemGbmGLESContext()
+: m_pGLContext{EGL_PLATFORM_GBM_MESA, "EGL_MESA_platform_gbm"}
```

Comment:
This breaks Rockchip as libmali do not expose `EGL_MESA_platform_gbm` client extension

Reply:
It does not have to. If it's not available, it will use plain `eglGetDisplay`. It supports that, right?

Reply:
`eglGetDisplay` should work with a gbm only version of libmali, the x11 or wayland driver used in debian/ubuntu will however stop working with Kodi gbm windowing after this change.
For LibreELEC and Rockchip this is no problem as we ship the gbm only version of libmali, but for Amlogic and Allwinner a multi platform libmali is used and will stop working with a strict check.

Please remove the strict check for `EGL_MESA_platform_gbm` or also allow the EGL 1.4, `EGL_EXT_platform_base` and `EGL_KHR_platform_gbm` combo to use the egl platform api.

Reply:
> This breaks Rockchip as libmali do not report EGL_MESA_platform_gbm client extension

Just that I get this right: It does not report the extension, but it does support it?

> eglGetDisplay should work with a gbm only version of libmali, the x11 or wayland driver used in debian/ubuntu will however stop working with Kodi gbm windowing after this change.

Hm why is that? Theoretically it should be able to detect from the native pointer what win system it is dealing with. At least mesa does that.

Reply:
> Just that I get this right: It does not report the extension, but it does support it?

Yes, the `EGL_KHR_platform_gbm` and the `EGL_MESA_platform_gbm` are pretty much the same extension and use the same platform constant `0x31D7`, the first is targeted for the EGL 1.5 spec and the other is for EGL 1.4 spec, libmali just reports `EGL_KHR_platform_gbm` instead of `EGL_MESA_platform_gbm` for some reason.
Same goes for `EGL_KHR_platform_wayland` (1.5) and `EGL_EXT_platform_wayland` (1.4), both use `0x31D8`.

Example client extensions for different libmali versions:
- gbm: `EGL_EXT_platform_base EGL_KHR_platform_gbm`
- wayland: `EGL_EXT_platform_base EGL_KHR_platform_gbm EGL_KHR_platform_wayland`
- x11: `EGL_EXT_platform_base EGL_KHR_platform_x11 EGL_KHR_platform_gbm`

> Hm why is that? Theoretically it should be able to detect from the native pointer what win system it is dealing with. At least mesa does that.

No idea, libmali do not seem to have such detections and the caller has to be explicit using `EGL_EXT_platform_base` calls or the default platform is used (wayland/x11).

Reply:
> No idea, libmali do not seem to have such detections and the caller has to be explicit using EGL_EXT_platform_base calls or the default platform is used (wayland/x11).

Can you show me where to download the version you are referring to?

Reply:
For Rockchip: https://github.com/rockchip-linux/libmali/tree/rockchip/lib/arm-linux-gnueabihf
For Amlogic: https://github.com/superna9999/meson_gx_mali_450/releases/download/for-4.15/buildroot_openlinux_kernel_4.9_20180131_mali.tar.gz

Reply:
> For Amlogic: https://github.com/superna9999/meson_gx_mali_450/releases/download/for-4.15/buildroot_openlinux_kernel_4.9_20180131_mali.tar.gz

OK thanks - as far as I can see that does not support wayland/x11. I'm looking for the driver you're referring to here: "the x11 or wayland driver used in debian/ubuntu will however stop working"

Reply:
Rockchip wayland: https://github.com/rockchip-linux/libmali/blob/rockchip/lib/arm-linux-gnueabihf/libmali-utgard-450-r7p0-wayland.so
Rockchip x11: https://github.com/rockchip-linux/libmali/blob/rockchip/lib/arm-linux-gnueabihf/libmali-utgard-450-r7p0.so
Allwinner H3 wayland: https://github.com/mosajjal/r6p2/blob/master/libwayland_for_mali/h3/lib_wayland/libMali.so